controller/linodemachine_controller_helpers.go

```diff
@@ -94,6 +94,14 @@ func (r *LinodeMachineReconciler) newCreateConfig(ctx context.Context, machineSc
 		createConfig.RootPass = uuid.NewString()
 	}
 
+	// add public interface to linode (eth0)
+	iface := &linodego.InstanceConfigInterfaceCreateOptions{
+		Purpose: linodego.InterfacePurposePublic,
+		Primary: true,
+	}
+	createConfig.Interfaces = append(createConfig.Interfaces, *iface)
```

Comment:
nit: If we're going to assign a pointer to `iface` and then immediately deference it, we can just assign the struct itself to `iface`

Reply:
Updated